### PR TITLE
Detail view

### PR DESCRIPTION
**Pull request by bart-de-jonge**
Added first version of detail view with handlers and functionality

### CODE REVIEW

File: src/main/java/control/ControllerManager.java [hunk 1]
```diff
@@ -1,19 +1,33 @@
...
     private RootPane rootPane;
 
     @Getter
     private TimelineController timelineControl;
 
+    @Getter
+    private DetailViewController detailViewController;
+
+    @Getter
+    private ShotBlock activeBlock;
```
Comment (martijnjanssen) [inline src/main/java/control/ControllerManager.java]: This is a `ShotBlock`, why is the variable not called `activeShotBlock`?

File: src/main/java/control/DetailViewController.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+    /**
+     * Init the begincount handlers.
+     */
+    private void initBeginCount() {
+        detailView.setBeginCount(0);
+
+        detailView.getBeginCountField().textProperty().addListener((
+                observable, oldValue, newValue) -> {
```
Comment (martijnjanssen) [inline src/main/java/control/DetailViewController.java]: The starting `(` is part of `observable, oldValue, newValue)`, this should be together.

File: src/main/java/control/DetailViewController.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                if (manager.getActiveBlock() != null) {
+                    int newVal = !newValue.isEmpty() ? Integer.parseInt(newValue) : 0;
+                    manager.getActiveBlock().setBeginCount(newVal);
+                    if (manager.getActiveBlock() instanceof CameraShotBlock) {
+                        ((CameraShotBlock) manager.getActiveBlock()).getShot()
+                                .setBeginCount(newVal);
```
Comment (martijnjanssen) [inline src/main/java/control/DetailViewController.java:44]: This indentation cannot be correct.

File: src/main/java/control/DetailViewController.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+    /**
+     * Init the description handlers.
+     */
+    private void initDescription() {
+        detailView.setDescription("");
+
+        detailView.getDescriptionField().textProperty().addListener((
+                observable, oldValue, newValue) -> {
```
Comment (martijnjanssen) [inline src/main/java/control/DetailViewController.java]: The `(` situation again.

File: src/main/java/control/DetailViewController.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+            });
+    }
+
+    /**
+     * Init the name handler.
+     */
+    private void initName() {
+        detailView.setName("");
+
+        detailView.getNameField().textProperty().addListener((
```
Comment (martijnjanssen) [inline src/main/java/control/DetailViewController.java]: The `(` situation again.

File: src/main/java/control/TimelineController.java [hunk 6]
```diff
@@ -36,16 +36,20 @@
     // TODO: Replace Camera Type and Scripting Project when XML functionality is available
     private final CameraType defType = new CameraType("AW-HE130 HD PTZ", "It's an IP Camera", 0.5);
 
-    // Placeholder project in lieu of XML loading
     @Getter
-    private ScriptingProject scriptingProject = new ScriptingProject("BOSS Project", 1.0);
+    private ControllerManager manager;
```
Comment (martijnjanssen) [inline src/main/java/control/TimelineController.java]: A variable name as `controllerManager` would better represent that class that is in the value.

File: src/main/java/control/TimelineController.java [hunk 7]
```diff
@@ -115,7 +123,7 @@ public void save() {
         File file = fileChooser.showSaveDialog(rootPane.getPrimaryStage());
         if (file != null) {
             System.out.println(file.toString());
-            scriptingProject.write(file);
+            project.write(file);
```
Comment (martijnjanssen) [inline src/main/java/control/TimelineController.java]: This newline is not needed.

**Comment by Markvanderuit** [comment]
Code looks good and application works on my machine, but it doesn't pass yet :(

File: src/main/java/gui/ShotBlock.java [hunk 8]
```diff
@@ -40,7 +62,10 @@ public ShotBlock(RootCenterArea rootCenterArea, double beginCount, double endCou
      * @param recompute - should we recompute after setting
      */
     public void setBeginCount(double count, boolean recompute) {
-        this.beginCount = count;
+
```
Comment (martijnjanssen) [inline src/main/java/gui/ShotBlock.java]: This is not needed.

File: src/main/java/control/DetailViewController.java (hunk 3 above)
Comment (bart-de-jonge) [inline src/main/java/control/DetailViewController.java:44]: Yes it is according to our checkstyle rules

**Comment by martijnjanssen** [comment]
Changes look good, all the things that were not correct are now fixed! Thanks, merging this.

**Pull request merged**